internal/worker/worker_test.go

```diff
@@ -49,42 +59,111 @@ func TestNew(t *testing.T) {
 		if !errors.Is(err, w.err) {
 			return errors.Errorf("got error = %v, want %v", err, w.err)
 		}
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got = %v, want %v", got, w.want)
+
+		egComparator := func(x, y errgroup.Group) bool {
+			return reflect.DeepEqual(x, y)
+		}
+		atomicValueComparator := func(x, y atomic.Value) bool {
+			return reflect.DeepEqual(x.Load(), y.Load())
+		}
+		want := w.want.(*worker)
+
+		queueOpts := []comparator.Option{
+			comparator.AllowUnexported(*(want.queue.(*queue))),
+			comparator.Comparer(func(x, y chan JobFunc) bool {
+				return len(x) == len(y)
+			}),
+			comparator.Comparer(egComparator),
+			comparator.Comparer(atomicValueComparator),
+		}
+		opts := []comparator.Option{
+			comparator.AllowUnexported(*want),
+			comparator.Comparer(func(x, y Queue) bool {
+				return comparator.Equal(x, y, queueOpts...)
+			}),
+			comparator.Comparer(egComparator),
+			comparator.Comparer(atomicValueComparator),
+		}
+		if diff := comparator.Diff(want, got, opts...); diff != "" {
+			return errors.New(diff)
 		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           opts: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           opts: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		{
+			name: "return worker without option",
+			want: want{
+				want: &worker{
+					name:       "worker",
+					limitation: 10,
+					eg:         errgroup.Get(),
+					running: func() atomic.Value {
+						v := new(atomic.Value)
+						v.Store(false)
+						return *v
+					}(),
```

Comment:
```suggestion
					running: func() (v atomic.Value) {
						v.Store(false)
						return v
					}(),
```
@hlts2 commented to me #606

Reply:
I think both are the same. anyway I fixed it.

Reply:
yes. I think so.